Assets/Scripts/FireSound.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FireSound : MonoBehaviour
+{
+    FireController fireController;
+    private AudioSource audioSource;
+    [SerializeField] private float addedVolumeMultiplier = 0.5f;
+    [SerializeField] private AudioClip fireClip;
+    float maxDistanceToPlayer = 20f;
+    [SerializeField] float minimumFireSoundVolume = 0.3f;
+    float fireSoundUpdateInterval = 0.2f;
+
+    void Start()
+    {
+        audioSource = gameObject.AddComponent<AudioSource>() as AudioSource;
+        audioSource.loop = true;
+        audioSource.clip = fireClip;
+        audioSource.Play();
+        fireController = GameObject.Find("MapManager").GetComponent<FireController>();
+        InvokeRepeating("AdjustSound", 0f, fireSoundUpdateInterval);
+    }
+
+    void AdjustSound()
+    {
+        float distanceToPlayer = maxDistanceToPlayer;
+        for (int y = 0; y < fireController.mapSizeY; y++)
+        {
+            for (int x = 0; x < fireController.mapSizeX; x++)
+            {
+                if(fireController.map[x, y].GetComponent<TileFire>().fireResistanceCurrent <= 0 && fireController.map[x, y].GetComponent<TileFire>().fireDuration > 0)
+                {
+                    float distanceToThisTile = Vector3.Distance(fireController.map[x, y].transform.position, transform.position);
+                    if (distanceToThisTile < distanceToPlayer)
+                        distanceToPlayer = distanceToThisTile;
+                }    
+            }
+        }
```

Comment:
This all can be done so much simpler. Just get all the tiles from MapManager, filter the ones, which are burning (with `List.FindAll()`) and find the closest one from the filtered list in a way, which we use in BachelorProject. You don't have to create an extension function like there, but at least the way it's done is much more efficient there. Then take the distance and apply it to the `distanceToPlayer` variable